WebDriverAgentLib/Routing/FBSession.m

```diff
@@ -181,6 +181,9 @@ - (XCUIApplication *)activeApplication
       if ([FBConfiguration shouldRespectSystemAlerts]
           && [XCUIApplication.fb_systemApplication descendantsMatchingType:XCUIElementTypeAlert].count > 0) {
         return XCUIApplication.fb_systemApplication;
+      } else if ([FBConfiguration shouldRespectHalfModals]
+                 && [XCUIApplication.fb_systemApplication fb_descendantsMatchingProperty:@"name" value:@"SBTransientOverlayWindow" partialSearch:false].count > 0) {
```

Comment:
not sure if halfModals is the correct name for such case. Perhaps `repsectTransientOverlayWindows` would better reflect its purpose?

Reply:
I think that using native xctest methods for lookup would be faster here (e.g. descendantsMatchingType ->matchingPredicate)